go/vt/vtorc/logic/topology_recovery.go

```diff
@@ -254,7 +254,7 @@ func recoverDeadPrimary(ctx context.Context, analysisEntry inst.ReplicationAnaly
 	}
 
 	// Read the tablet information from the database to find the shard and keyspace of the tablet
-	tablet, err := inst.ReadTablet(analysisEntry.AnalyzedInstanceKey)
+	tablet, _ := inst.ReadTablet(analysisEntry.AnalyzedInstanceKey)
```

Comment:
@GuptaManan100 it seems wrong to ignore the error here. Should be exist the func altogether if we aren't able to read the tablet?

Reply:
Yes, we should exit the function altogether. We can't proceed without the tablet's information. We'll get a panic when we try to read the keyspace and shard from this tablet information reference. If it was ignored before, that's a bug.

Reply:
```golang
if err != nil {
    return false, nil, err
}
```
is all we need after that line.

Reply:
Thank you @GuptaManan100, I'll port this change on `release-16.0`

Reply:
The change made in 42eb6db will fix https://github.com/vitessio/vitess/issues/12509